app/controllers/katello/api/v2/repository_sets_controller.rb

```diff
@@ -122,7 +130,22 @@ def index_relation
 
       relation = relation.where(Katello::Content.table_name => {:name => params[:name]}) if params[:name].present?
       relation = relation.redhat unless ::Foreman::Cast.to_bool(params[:with_custom])
-      relation
+      index_relation_with_activation_key_overrides(relation)
+    end
+
+
+    def index_relation_with_activation_key_overrides(relation)
+      return relation if @activation_key.nil?
```

Comment:
I kind of prefer using Rails' `.blank?` over `.nil?` since it also covers empty strings, so it feels safer